### PR TITLE
feat: Write code challange sub array with given sum

### PR DESCRIPTION
**Pull request by octavioeac**

### CODE REVIEW

File: JavaFolder/code/src/main/java/xyz/krakenkat/app/interview/SubArrayWithGivenSum.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
+package xyz.krakenkat.app.interview;
```
Comment (droidark) [inline JavaFolder/code/src/main/java/xyz/krakenkat/app/interview/SubArrayWithGivenSum.java:1]: 1. The repository language is English, please either remove the Spanish comments or translate them.
2. Add a full description of the code.

File: JavaFolder/code/src/main/java/xyz/krakenkat/app/interview/SubArrayWithGivenSum.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+public class SubArrayWithGivenSum {
+
+	public static void main(String[] args) {
```
Comment (droidark) [inline JavaFolder/code/src/main/java/xyz/krakenkat/app/interview/SubArrayWithGivenSum.java:5]: Remove the main method. Please add the required tests for your code.

File: JavaFolder/code/src/main/java/xyz/krakenkat/app/interview/SubArrayWithGivenSum.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+		test(new int[] {
+				10,3,5,8,6,12,20,15,31
+		}, 31);
+		
+	}
+
+	
+	public static void test(int[] array, int num) {
```
Comment (droidark) [inline JavaFolder/code/src/main/java/xyz/krakenkat/app/interview/SubArrayWithGivenSum.java:14]: Remove the static method. The static method will cause issues in the unit tests.

Comment (droidark) [inline JavaFolder/code/src/main/java/xyz/krakenkat/app/interview/SubArrayWithGivenSum.java:14]: What is the return value?

**Comment by morelos1904** [comment]
Cual es el problema ? 
 Buscar un par de numeros que sumados te den X resultado ? O es otro el problema? @octavioeac

File: JavaFolder/code/src/main/java/xyz/krakenkat/app/interview/SubArrayWithGivenSum.java (hunk 1 above)
Comment (droidark) [reply JavaFolder/code/src/main/java/xyz/krakenkat/app/interview/SubArrayWithGivenSum.java:1]: The developer made the promise to update the code. I trust him

File: JavaFolder/code/src/main/java/xyz/krakenkat/app/interview/SubArrayWithGivenSum.java (hunk 3 above)
Comment (droidark) [reply JavaFolder/code/src/main/java/xyz/krakenkat/app/interview/SubArrayWithGivenSum.java:14]: The developer made the promise to update the code. I trust him

Comment (droidark) [reply JavaFolder/code/src/main/java/xyz/krakenkat/app/interview/SubArrayWithGivenSum.java:14]: The developer made the promise to update the code. I trust him

**Review by droidark — APPROVED** [review]
The developer made the promise to update the code. I trust him

File: JavaFolder/code/src/main/java/xyz/krakenkat/app/interview/SubArrayWithGivenSum.java (hunk 2 above)
Comment (octavioeac) [reply JavaFolder/code/src/main/java/xyz/krakenkat/app/interview/SubArrayWithGivenSum.java:5]: Sure!

**Comment by octavioeac** [comment]
Es el siguiente problema que ví en un video de youtube: @morelos1904 No he puesto la descripción, pero cuando tenga chance la pondre.  Subiré más problemas de los cuales me he enfrentado en las entrevitas.
https://www.youtube.com/watch?v=Aa1j__CUzXo&list=PL1w8k37X_6L-bCZ3m0FFBZmRv4onE7Zjl&index=2

**Pull request merged**